src/fdosecrets/FdoSecretsPlugin.cpp

```diff
@@ -33,9 +32,9 @@ QPointer<FdoSecretsPlugin> g_fdoSecretsPlugin;
 
 FdoSecretsPlugin::FdoSecretsPlugin(DatabaseTabWidget* tabWidget)
     : m_dbTabs(tabWidget)
+    , m_dbus(*this, QDBusConnection::sessionBus())
```

Comment:
Dereferencing `this` in this particular case should be OK because it is not accessed in `DBusMgr`'s constructor. But I agree with you that it's better to get rid of this usage.

A global instance for the plugin makes sense for me, and we are on the same page about decoupling it from the GUI (I've added this to my TODO to further decouple the plugin and the GUI, and eventually have the plugin instance as the only bridge between the GUI and other fdosecrets code). But I don't feel it solves this particular problem, though. I feel the global instance should only be accessed by the GUI part of the program, and IMHO `DBusMgr` is a component of the plugin, and it should be agnostic to whether the plugin instance is global or something else.

Anyway, future plan aside, for this, I think simply have `DBusMgr` emits signals on errors and `FdoSecretsPlugin` 
 can connect to the signal in the constructor. This should get rid of the pointer/reference in `DBusMgr`.

Reply:
Excellent plan